Add dynamic note selection that is responsive to changes in required fees. (#689)
zcash/librustzcash: zcash_client_backend/CHANGELOG.md

  - `WalletWrite::get_next_available_address`
  - `WalletWrite::put_received_transparent_utxo`
  - `impl From<prost::DecodeError> for error::Error`

>   - `chain::error`: a module containing error types type that that can occur only

Reviewer: ```suggestion
  - `chain::error`: a module containing error types that that can occur only
```

>   - `chain::error`: a module containing error types type that that can occur only
>     in chain validation and sync have been separated out from errors related to
>     other wallet operations.

Reviewer: ```suggestion
  - `chain::error`: a module containing error types that can occur only in chain
    validation and sync, which have been separated out from errors related to
    other wallet operations.
```

  - `TransactionBalance`
  - `BasicFixedFeeChangeStrategy` - a `ChangeStrategy` implementation that
    reproduces current wallet change behavior

>   - `fixed` a new module containing of change selection strategies for the

Reviewer 2: ```suggestion
  - `fixed`, a new module containing of change selection strategies for the
```
(or `:` as was used above)

  `zcash_client_backend::data_api::chain::testing::MockBlockSource` module.
- `zcash_client_backend::data_api::chain::{validate_chain, scan_cached_blocks}`
  have altered parameters and result types. The latter have been modified to
  return`zcash_client_backend::data_api::chain::error::Error` instead of
  abstract error types. This new error type now wraps the errors of the block
  source and wallet database to which these methods delegate IO operations
  directly, which simplifies error handling in cases where callback functions
  are involved.
- `zcash_client_backend::data_api::error::ChainInvalid` has been moved to
  `zcash_client_backend::data_api::chain::error`.
- The error type of `zcash_client_backend::data_api::wallet::decrypt_and_store_transaction`
  has been changed; the error type produced by the provided `WalletWrite` instance is
  returned directly.

### Deprecated
- `zcash_client_backend::data_api::wallet::create_spend_to_address` has been
  deprecated. Use `zcash_client_backend::data_api::wallet::spend` instead. If
  you wish to continue using `create_spend_to_address`, note that the arguments

>   to the function has been modified to take a unified spending key instead of a

Reviewer: ```suggestion
  to the function have been modified to take a unified spending key instead of a
```

>   Sapling extended spending key, and now also requires a `min_confirmations`

Reviewer: ```suggestion
  Sapling extended spending key, and it now also requires a `min_confirmations`
```

>   to the function has been modified to take a unified spending key instead of a
>   Sapling extended spending key, and now also requires a `min_confirmations`

Reviewer: ```suggestion
  to the function have been modified to take a unified spending key instead of a
  Sapling extended spending key, and it now also requires a `min_confirmations`
```
